### ISSUE TITLE
[java] PMD 7.0.0-rc3 JUnitAssertionsShouldIncludeMessage reports assertThat from hamcrest

### ISSUE THREAD

**Issue by dant3** [pmd/pmd#4585]
**Affects PMD Version:** 7.0.0-rc3

**Rule:** [JUnitAssertionsShouldIncludeMessage](https://docs.pmd-code.org/pmd-doc-7.0.0-rc3/pmd_rules_java_bestpractices.html#junitassertionsshouldincludemessage)

https://docs.pmd-code.org/pmd-doc-7.0.0-rc3/pmd_rules_java_bestpractices.html#junitassertionsshouldincludemessage

**Description:**

**Code Sample demonstrating the issue:**

```java
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.nullValue;
import static org.hamcrest.Matchers.is;

...

Object foo = null;
assertThat(foo, is(nullValue()));

```

**Expected outcome:**

PMD reports a violation at line ```assertThat(*, *)```, but that's wrong. That's a false positive. Hamcrest matchers provide message on their own. Notice that assertThat also comes from Hamcrest, not JUnit.

**Running PMD through:** *Gradle*

**Comment by jsotuyod** [comment]
This is also true for the (deprecated) JUnit 4 `assertThat` methods… even if they provide an overload that accepts a `String reason`, all `Matcher` are `SelfDescribing` and can create reason messages on their own.

TestNG have no `assertThat` method, so it may be simply removing this line… https://github.com/pmd/pmd/blob/e8dbb54cb5ed682d9dfd4f54895f4bbd73be728e/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/JUnitAssertionsShouldIncludeMessageRule.java#L25

Tests around  `assertThat` would need to be updated accordingly.
